src/cargo/sources/git/source.rs

```diff
@@ -76,6 +82,14 @@ fn ident(id: &SourceId) -> String {
     format!("{}-{}", ident, short_hash(id.canonical_url()))
 }
 
+fn ident_shallow(id: &SourceId, is_shallow: bool) -> String {
```

Comment:
That's good to know! When working on a topic I might naturally be able to 'share' the knowledge by not only documenting what's new, but also documenting what I think should be said about what's there. At worst it can be deleted, at best it can be improved as part of the review.

Addressed in 5716fe521cd49fe71d17723fd6ea0292ce626533.